src/System/Validation/Code.cs

```diff
@@ -198,6 +221,38 @@ public static bool Validate(bool state, uint? tagId, string errorMessage, params
 		}
 
 
+		/// <summary>
+		/// Checks that the enumerable argument is not null, is not empty, and does not contain nulls
+		/// </summary>
+		/// <remarks>Be careful to not pass enumerables that can be enumerated only once</remarks>
+		/// <typeparam name="T">The type of the enumerable</typeparam>
+		/// <param name="argumentValue">The argument value.</param>
+		/// <param name="argumentName">Name of the argument.</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed</param>
+		/// <returns>True if the argument <paramref name="argumentValue"/> is not null, is not empty, and does not contain nulls; false otherwise.</returns>
+		public static bool ValidateAnyAndAllNotNull<T>(IEnumerable<T> argumentValue, string argumentName, uint? tagId) 
+			where T : class
+		{
+			if (!ValidateArgument(argumentValue, argumentName, tagId))
+			{
+				return false;
+			}
+
+			if (!argumentValue.Any() || argumentValue.Any(x => x == null))
```

Comment:
I don't like double iteration over `IEnumerable` here, strictly speaking, we should've used `ICollection<T>` as a parameter type here, but that that would require changes in downstream repositories.

What do you think about replacing it with good old `foreach` to avoid double iteration?

Reply:
Agree the double iteration isn't good. For this method we could use an `ICollection<T>` as there won't be any dependencies on it yet? Or if we want to keep it consistent using `IEnumerable<T>` I'll replace it with a `foreach`.

Reply:
I would probably go with `ICollection<T>` since after calling this validation method we would probably need iterate over a collection again to get values.

Reply:
Updated to use `ICollection<T>`